### ISSUE TITLE
Parameter validator classes cannot be inner classes

### ISSUE THREAD

**Issue by Zero3** [cbeust/jcommander#226]
Trying to implement an IParameterValidator class as an inner class causes a runtime exception like the following example:

    Exception in thread "main" com.beust.jcommander.ParameterException: Can't instantiate validator:java.lang.InstantiationException: dk.zero3.project.Settings$PortValidator
            at com.beust.jcommander.ParameterDescription.validateParameter(ParameterDescription.java:307)
            at com.beust.jcommander.ParameterDescription.validateParameter(ParameterDescription.java:269)
            at com.beust.jcommander.ParameterDescription.addValue(ParameterDescription.java:238)
            at com.beust.jcommander.ParameterDescription.addValue(ParameterDescription.java:210)
            at com.beust.jcommander.JCommander.processFixedArity(JCommander.java:864)
            at com.beust.jcommander.JCommander.processFixedArity(JCommander.java:846)
            at com.beust.jcommander.JCommander.parseValues(JCommander.java:718)
            at com.beust.jcommander.JCommander.parse(JCommander.java:279)
            at com.beust.jcommander.JCommander.parse(JCommander.java:262)
            at com.beust.jcommander.JCommander.<init>(JCommander.java:207)
            at dk.zero3.project.Project.main(Project.java:17)`

It would be nice to be able to use an inner class, since you then may embed the validators inside your settings class, keeping things in the same place. Example:

    public class Settings
    {
    	@Parameter(names = "-port", validateWith = PortValidator.class, description = "Webserver port")
    	public Integer port;

        public static class PortValidator implements IParameterValidator
        {
        	@Override
        	public void validate(String name, String value) throws ParameterException
        	{
        		int port = Integer.parseInt(value);

        		if (port < 1 || port > 65535)
        		{
        			throw new IllegalArgumentException("Invalid port number specified. Must be in the range 1 - 65535.");
        		}
        	}
        }
    }

**Comment by cbeust** [comment]
That's surprising, it should work. Are you sure that

- Your validator has a parameterless constructor.
- It's public.
- Its enclosing class is public.

?

**Comment by Zero3** [comment]
Weird, it works now that I tried my minimal example again. Must have been a mistake on my end. Sorry! :)

**Comment by kevinarpe** [comment]
Just an FYI: I triggered this same issue accidentally today.  I had my validator as an innner/nested class, but forgot to include 'static'.  Doh!  Once I added static back, it worked.  Maybe a nice to have... if the instantiation fails... check if class is non-static and give a nice exception message... "Dear Developer: Maybe you are using an inner / nested class but forget 'static' keyword."  (First problems!)
